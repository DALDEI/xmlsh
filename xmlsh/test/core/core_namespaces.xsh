# Test of core namespaces support
. ../common

# declare 2 namespaces
declare namespace t1=http://www.example.org/test1
declare namespace t2=http://www.example.org/test2
declare namespace
# Turn off indentation for this test
set +indent

echo loc() <[ <t1:test><t2:test>Test</t2:test></t1:test> ]>

# undeclare t2
declare namespace t2=
declare namespace

#echo Expected Error 
#echo <[ <t1:test><t2:test>Test</t2:test></t1:test> ]>

# OK
echo loc()  <[ <t1:test><test>Test</test></t1:test> ]>

# Unset namespaces
declare namespace t2=
declare namespace t1=

#try with inband variable assignment
# DEPRECIATED
# xmlns:t3=http://www.example.org/test3 eval 'echo <[ <t3:test><test>Test</test></t3:test> ]>'
{ 
  declare namespace t3=http://www.example.org/test3
  eval 'echo loc() <[ <t3:test><test>Test</test></t3:test> ]>'
}
# Default namespace
declare namespace http://test.xmlsh.org/ns
echo loc() <[ <foo/> ]>
#
declare namespace ""
echo loc() <[ <foo/> ]>

