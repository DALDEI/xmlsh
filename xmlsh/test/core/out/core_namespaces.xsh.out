t1=http://www.example.org/test1
t2=http://www.example.org/test2
[11] <t1:test xmlns:t1="http://www.example.org/test1"><t2:test xmlns:t2="http://www.example.org/test2">Test</t2:test></t1:test>
t1=http://www.example.org/test1
[21] <t1:test xmlns:t1="http://www.example.org/test1"><test>Test</test></t1:test>
[1] <t3:test xmlns:t3="http://www.example.org/test3"><test>Test</test></t3:test>
[36] <foo xmlns="http://test.xmlsh.org/ns"/>
[39] <foo/>
